src/clib/cool1d_multi_g.F

```diff
@@ -679,79 +1034,520 @@ subroutine cool1d_multi_g(
+               gasMgSiO3  (i) = fv2k * sgMgSiO3  (i)
+               gasAC      (i) = fv2k * sgAC      (i)
+               endif
+               if (idspecies .gt. 1) then
+               gasSiM     (i) = fv2k * sgSiM     (i)
+               gasFeM     (i) = fv2k * sgFeM     (i)
+               gasMg2SiO4 (i) = fv2k * sgMg2SiO4 (i)
+               gasFe3O4   (i) = fv2k * sgFe3O4   (i)
+               gasSiO2D   (i) = fv2k * sgSiO2D   (i)
+               gasMgO     (i) = fv2k * sgMgO     (i)
+               gasFeS     (i) = fv2k * sgFeS     (i)
+               gasAl2O3   (i) = fv2k * sgAl2O3   (i)
+               endif
+               if (idspecies .gt. 2) then
+               gasreforg  (i) = fv2k * sgreforg  (i)
+               gasvolorg  (i) = fv2k * sgvolorg  (i)
+               gasH2Oice  (i) = fv2k * sgH2Oice  (i)
+               endif
+
+               if (idspecies .gt. 0) then
+               gasgr_tMgSiO3  (i) = gasMgSiO3  (i) * fac
+               gasgr_tAC      (i) = gasAC      (i) * fac
+               endif
+               if (idspecies .gt. 1) then
+               gasgr_tSiM     (i) = gasSiM     (i) * fac
+               gasgr_tFeM     (i) = gasFeM     (i) * fac
+               gasgr_tMg2SiO4 (i) = gasMg2SiO4 (i) * fac
+               gasgr_tFe3O4   (i) = gasFe3O4   (i) * fac
+               gasgr_tSiO2D   (i) = gasSiO2D   (i) * fac
+               gasgr_tMgO     (i) = gasMgO     (i) * fac
+               gasgr_tFeS     (i) = gasFeS     (i) * fac
+               gasgr_tAl2O3   (i) = gasAl2O3   (i) * fac
+               endif
+               if (idspecies .gt. 2) then
+               gasgr_treforg  (i) = gasreforg  (i) * fac
+               gasgr_tvolorg  (i) = gasvolorg  (i) * fac
+               gasgr_tH2Oice  (i) = gasH2Oice  (i) * fac
+               endif
+
+               endif
+
+            endif ! idspecies
+
             endif
          enddo
 
+!        if (idspecies .eq. 0) then
+!           do itd = 1, gr_N(2)
+!              do i = is+1, ie + 1
+!              if(itmask_metal(i)) then
+!                 altot(itd,i)=
+!    &                (SN0_kpFeM    (4*itd,1)*SN0_fFeM    (1)
+!    &                /SN0_r0FeM    (3    ,1)
+!    &                +SN0_kpMg2SiO4(4*itd,1)*SN0_fMg2SiO4(1)
+!    &                /SN0_r0Mg2SiO4(3    ,1)
+!    &                +SN0_kpMgSiO3 (4*itd,1)*SN0_fMgSiO3 (1)
+!    &                /SN0_r0MgSiO3 (3    ,1)
+!    &                +SN0_kpFeS    (4*itd,1)*SN0_fFeS    (1)
+!    &                /SN0_r0FeS    (3    ,1)
+!    &                +SN0_kpreforg (4*itd,1)*SN0_freforg (1)
+!    &                /SN0_r0reforg (3    ,1)
+!    &                +SN0_kpvolorg (4*itd,1)*SN0_fvolorg (1)
+!    &                /SN0_r0volorg (3    ,1)
+!    &                +SN0_kpH2Oice (4*itd,1)*SN0_fH2Oice (1)
+!    &                /SN0_r0H2Oice (3    ,1))
+!    &                  * metal(i,j,k) / d(i,j,k)
+!              endif
+!              enddo
+!           enddo
+!        endif
+
```

Comment:
Do we need to keep this around? Or if this is an idea that should be explored in the future, can you add a clarifying comment about what you were trying to do here?

Reply:
This calculates dust opacity when idspecies = 0 (default).
Originally dust opacity is given by the simple function (4e-4 * Tdust ^2 in calc_tdust_1d_g.F).
I updated this to the more complicated one but commented out because it needs too many double calculations. 
So we can delete this for now. I'll give the dust opacity with a more simple table in a future update.